### ISSUE TITLE
CamelCloudEventXXX not converted to CE header

### ISSUE THREAD

**Issue by nicolaferraro** [apache/camel-k-runtime#177]
I'm setting header CamelCloudEventType to "something" during an integration, but it's not converted to `ce-type` going out.

In general, should we tell users to set "ce-type" or "CamelCloudEventType" when they want to override that field?

The same question for CE extensions. Should they use "ce-author" or "CamelCloudEventAuthor"?
"ce-author" may seem obvious, but extensions are not always mapped to http headers, especially when using structured json output.

Dump attached:

```
POST / HTTP/1.1
Host: dumper.default.svc
User-Agent: Vert.x-WebClient/3.8.1
Content-Length: 3
Accept-Encoding: gzip
Ce-Author: Nicola Ferraro
Ce-Chat: 158584902
Ce-Id: ID-bot-source-r2tnr-74c4cd6d48-68zkg-1572176653802-0-1
Ce-Source: camel-source:default/bot-source
Ce-Specversion: 0.3
Ce-Time: 2019-10-27T11:44:13.8Z
Ce-Type: org.apache.camel.event
Content-Type: text/plain
Forwarded: for=172.17.0.27;proto=http, for=172.17.0.10
K-Proxy-Request: activator
X-B3-Sampled: 1
X-B3-Spanid: 0700a32370b41102
X-B3-Traceid: 9035b130ad9a32340700a32370b41102
X-Envoy-Decorator-Operation: dumper-xbn7f.default.svc.cluster.local:80/*
X-Envoy-Expected-Rq-Timeout-Ms: 600000
X-Envoy-Internal: true
X-Forwarded-For: 172.17.0.27, 172.17.0.10, 172.17.0.11
X-Forwarded-Proto: http
X-Request-Id: fd6d15cd-e281-94cf-961d-ab0fc70d7f1b

```

**Comment by lburgazzoli** [comment]
Should be fixed by #178.

**Comment by lburgazzoli** [comment]
`CamelCloudEvent*` are helpers for user to avoid caring about the cloud event spec version but of course they are limited to the known attributes. 

CloudEvent headers are the precedence over `CamelCloudEvent*` so assuming a route using CloudEvent spec  v0.3 like:

```java
from("direct:start")
    .setHeader("ce-type")).constant("fromCEHeader")
    .setHeader("CamelCloudEventType").constant("fromCamelHeader")
    .to("knative:endpoint/ep");
```

The cloud event type will be set to `fromCEHeader` 

Note that:
- camel-knative producer does not generate structured content but always binary content whereas the consumer can handle both.
- `CamelCloudEvent*` on consumer side are added **in addition** to the CloudEvent header recevived by the consumer
